lib/src/migrators/module/scope.dart

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Google LLC
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:sass_migrator/src/utils.dart';
+
+// The sass package's API is not necessarily stable. It is being imported with
+// the Sass team's explicit knowledge and approval. See
+// https://github.com/sass/dart-sass/issues/236.
+import 'package:sass/src/ast/sass.dart';
+
+import 'package:path/path.dart' as p;
+import 'package:source_span/source_span.dart';
+
+/// Keeps track of the scope of any members declared at the current level of
+/// the stylesheet.
+class Scope {
+  /// The parent of this scope, or null if this scope is global.
+  final Scope parent;
+
+  /// Variables defined in this scope.
+  ///
+  /// These are usually VariableDeclarations, but can also be Arguments from
+  /// a CallableDeclaration.
+  final variables = normalizedMap<SassNode /*VariableDeclaration|Argument*/ >();
+
+  /// Mixins defined in this scope.
+  final mixins = normalizedMap<MixinRule>();
+
+  /// Functions defined in this scope.
+  final functions = normalizedMap<FunctionRule>();
+
+  /// Nested imports from this scope.
+  ///
+  /// Note: For the purposes of migration, any imports that come after a rule
+  /// other than @import, @use, @forward, or @charset are considered nested, since
+  /// they can't be migrated to @use.
+  final nestedImports = <Uri>{};
+
+  Scope(this.parent);
```

Comment:
```suggestion
  Scope([this.parent]);
```

Making `parent` optional helps communicate that constructing a parentless `Scope` is valid and expected, and it makes the callsite where that happens cleaner.